Reverse Clipping Mask Stacking Placement

manongjohn opened the pull request:
To follow more in line with the logical stacking order (and other software), this PR changes the placement and stacking order of Clipping Masks.

Clipping Masks now need to be placed **above** (timeline)/**right** (xsheet) of the clipped column.
Clipping is done in order from bottom->top/left->right.

![image](https://github.com/user-attachments/assets/c43e47b8-08af-4cef-929d-82df5182ed43)

Note: Due to change in stacking order, `Render Mask` has been removed.

artisteacher commented:
I think this definitely will make it easier going back & forth between different programs.

Is the "render mask" option working as expected?

manongjohn commented:
> Is the "render mask" option working as expected?

It should be. Was working for me in my tests.

I suppose the difference here is that when the clipping masks were below the clipped column, the rendered mask was below it.  Now it's rendered above it.

artisteacher commented:
So I have 2 levels.  This vector star above:
<img width="341" alt="Screenshot 2024-12-10 at 10 06 57 PM" src="https://github.com/user-attachments/assets/436f7193-2d6d-424f-bd35-9d5aecf35940">
And this raster image below:
<img width="446" alt="Screenshot 2024-12-10 at 10 06 49 PM" src="https://github.com/user-attachments/assets/8d860604-c538-4869-9be2-c4520269d13d">

With the clipping mask enabled + render mask, the mask completely obscures the clipped area:
<img width="810" alt="Screenshot 2024-12-10 at 10 06 21 PM" src="https://github.com/user-attachments/assets/834465cb-022a-4a25-a75c-48ffc7fe8529">
Which makes the raster level pointless. If you keep the render mask option, shouldn't you be able to see both?

With the clipping mask inverted + render mask, the mask isn't visible:
<img width="883" alt="Screenshot 2024-12-10 at 10 06 02 PM" src="https://github.com/user-attachments/assets/db6d8493-6bb2-44cc-acf6-307caa68a90b">
Though if it were, I guess there might not be a point in using the mask to begin with...

manongjohn commented:
@artisteacher Render mask was only visible when using Invert, even before this change.  Seems to be a bug with invert and render.  Odd because I know I had that working at one point when I was testing the reversal.   Will fix

manongjohn commented:
Upon further reflection, I believe you are correct and `Render Mask` is no longer necessary.

Clipping + Render Mask = essentially hides the image below it behind the drawn mask.
Clipping + Invert Mask + Render Mask = the same as just drawing the mask on top of the image without clipping.

Unless someone else disagrees with this, I can completely remove the `Render Mask` option.

artisteacher commented:
I think it could still be helpful for a default clipping mask, as long as the mask displays under the clipped area. Completely agree on removing for an inverted mask.

manongjohn commented:
> I think it could still be helpful for a default clipping mask, as long as the mask displays under the clipped area. It's the equivalent of the "visible matte in" fx. Completely agree on removing for an inverted mask.

Displaying a mask under the clipped area breaks the stacking order of things and complicates the rendering to a degree. They can just as easily accomplish this by placing a copy of the column, without mask enabled, below the clipped column.

artisteacher commented:
Was just hoping to save a step or 2, but that would certainly work too.

manongjohn commented:
Updated and slightly fixed PR due to conflicts with the Alpha Lock PR after it was merged.  With that, updated Alpha Locked columns to ignore clipping masks found between it and the reference column.

I'll leave this up a few extra days in case anyone wants to retest.

artisteacher commented:
Working just fine on my end.

senoworks commented:
thats how it should be the clip must be on the top, like software painting did
